src/databricks/labs/ucx/hive_metastore/federation.py

```diff
@@ -159,18 +166,24 @@ def _external_hms(self) -> ExternalHmsInfo | None:
     def _split_jdbc_url(cls, jdbc_url: str) -> ExternalHmsInfo:
         # Define the regex pattern to match the JDBC URL components
         pattern = re.compile(
-            r'jdbc:(?P<db_type>[a-zA-Z0-9]+)://(?P<host>[^:/]+):(?P<port>\d+)/(?P<database>[^?]+)(\?user=(?P<user>[^&]+)&password=(?P<password>[^&]+))?'
+            r'jdbc:(?P<db_type>[a-zA-Z0-9]+)://(?P<host>[^:/?;]+)(:(?P<port>\d+))?(/(?P<database>[^?^;]+))?([?;](?P<parameters>.+))?'
         )
         match = pattern.match(jdbc_url)
         if not match:
             raise ValueError(f'Unsupported JDBC URL: {jdbc_url}')
 
+        params = {}
+        if match.group('parameters'):
+            params = dict(param.split('=') for param in re.split(r"[;&]", match.group('parameters')))
```

Comment:
```suggestion
        parameters_raw = match.group('parameters')
        if parameters_raw:
            params = dict(param.split('=') for param in re.split(r"[;&]", parameters_raw))
```

Is the `re.split` needed? Why not a "regular" split like the `param.split`?

Reply:
Because there is more than one optional spliter & and ;